src/conda_devenv/devenv.py

```diff
@@ -426,6 +471,9 @@ def load_yaml_dict(
 
     root_yaml = yaml.safe_load(rendered_contents)
 
+    # Drop constraints from the root yaml file.
+    root_yaml.pop("constraints", None)
```

Comment:
It's not clear why, maybe the reason can be added to the comment?

Would it make sense for the root yaml to declare constraints for dependencies listed in the devenv files it includes (it's the opposite direction of what is usual)?

Reply:
Hmm thinking more about it, actually we do not really need this line at all, removed. 👍